posthog/cdp/templates/knock/template_knock.py

```diff
@@ -0,0 +1,81 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionTemplate
+
+template: HogFunctionTemplate = HogFunctionTemplate(
+    status="beta",
+    id="template-knock",
+    name="Send events to Knock",
+    description="Send events to Knock",
+    icon_url="/static/services/knock.png",
+    hog="""
+if (empty(inputs.userId)) {
+    print('No User ID set. Skipping...')
+    return
+}
+
+let body := {
+	'type': 'track',
+	'event': event.name,
+	'userId': inputs.userId,
+	'properties': inputs.include_all_properties ? event.properties : {},
+	'messageId': event.uuid,
+    'timestamp': event.timestamp
+}
```

Comment:
Tabs vs spaces... Probably best to be consistent and stick to tabs.

Reply:
So I've been using 4 space indents everywhere including the hog code. (I adjusted this code as well)

In the [Salesforce template](https://github.com/PostHog/posthog/blob/master/posthog/cdp/templates/salesforce/template_salesforce.py) we are using 4 space indents in the python code but only 2 space indents in the hog code.

Should we have different indents for hog vs python code?

Reply:
4 spaces is fine, but you used tabs there, which give different spaces depending on who's looking at it.

Reply:
Perfect - I've replaced the tabs with spaces 👍